docs/devel/owners.md

```diff
@@ -0,0 +1,128 @@
+
+After a PR is submitted, the automated kubernetes PR robot will append a message to the PR indicating the owners
+that are required for the PR to be submitted.
+
+Subsequently, a user can also request the approval message from the robot by writing:
+
+```
+@k8s-bot approvers
+```
+
+into a comment.
+
+In either case, the automation replies with an annotation that indicates
+the owners required to approve.  The annotation is a comment that is applied to the PR.
+This comment will say:
+
+```
+Approval is required from <owner-a> OR <owner-b>, AND <owner-c> OR <owner-d>, AND ...
+```
+
+The set of required authors is drawn from the OWNERS files in the repository (see below).  For each file
+there should be multiple different OWNERS, these owners are listed in the `OR` clause(s). Because
+it is possible that a PR may cover different directories, with disjoint sets of OWNERS, a PR may require
+approval from more than one person, this is where the `AND` clauses come from.
+
+### Step Two: A PR is LGTM'd
+
+Once a PR is reviewed and LGTM'd it is eligible for submission.  However, for it to be submitted
+an owner for all of the files changed in the PR have to 'approve' the PR.  A user is an owner for a
+file if they are included in the OWNERS hierarchy (see below) for that file.
+
+Owner approval comes in two forms:
+
+   * An owner adds a comment to the PR saying "I approve"
+   * An owner is the original author of the PR
+
+In the case of a comment based approval, the same rules as for the 'lgtm' label apply.  If the PR is
+changed by pushing new commits to the PR, the previous approval is invalidated, and the owner(s) must
+approve again.  Because of this is recommended that PR authors squash their PRs prior to getting approval
+from owners.
+
+### Step Three: A PR is merged
+
+Once a PR is LGTM'd and all owners have approved, it is eligible for merge.  The merge bot takes care of
+the actual merging.
+
+## Design details
+
+We need to build new features into the existing github munger in order to accomplish this.  Additionally
+we need to add owners files to the repository.
+
+### Approval Munger
+
+We need to add a munger that adds comments to PRs indicating whose approval they require.  This munger will
+look for PRs that do not have approvers already present in the comments, or where approvers have been
+requested, and add an appropriate comment to the PR.
+
+
+### Status Munger
+
+GitHub has a [status api](https://developer.github.com/v3/repos/statuses/), we will add a status munger that pushes a status onto a PR of approval status.  This status will only be approved if the relevant
+approvers have approved the PR.
+
+### Requiring approval status
+
+Github has the ability to [require status checks prior to merging](https://help.github.com/articles/enabling-required-status-checks/)
+
+Once we have the status check munger described above implemented, we will add this required status check
+to our main branch as well as any release branches.
```

Comment:
I see little purpose in this. The bot won't care (since it will both be setting the status and doing the merge only if the status would be set) and if humans are doing things wrong we need to 'discuss' with the humans. When the build breaks and we have a 1-liner to fix it I WANT a human to be able to just push it.

If we want/need a munger which flags those cases where someone merged something by hand, we can do that, but I'd really like to not prevent people from doing useful things in a timely manor (and slapping their wrists if they mess up)